### PR TITLE
Uob 43 add promise

### PR DESCRIPTION
**Pull request by gvdent**

### CODE REVIEW

**Comment by gvdent** [comment]
@malamili

**Comment by malamili** [comment]
adds() is a bit confusing, or is this from some sort of convention? Would addPromise not be beter, then have function(entity, error) resolve in promise?

**Comment by malamili** [comment]
does the promise btw correctly reject on server error?

File: src/weaver.coffee [hunk 1]
```diff
@@ -54,10 +55,21 @@ class Weaver
     attributes[key] = value for key, value of data when not isEntity(value)
     relations[key]  = value.$id() for key, value of data when isEntity(value)
 
-    @channel.create({type, id:entity.$id(), attributes, relations})
+    createPromise = @channel.create({type, id:entity.$id(), attributes, relations})
```
Comment (malamili) [inline src/weaver.coffee]: does this correctly rejects?

**Comment by gvdent** [comment]
Added a test to see whether reject is called on disconnect. So if the server correctly rejects when an error is called, the promise returned by addPromise should reject as well.

**Pull request merged**